carma_wm/src/SignalizedIntersectionManager.cpp

```diff
@@ -465,4 +465,283 @@ namespace carma_wm
+          getTrafficSignal(curr_light_id, semantic_map);
+
+        if (curr_light == nullptr)
+        {
+          continue;
+        }
+
+        // reset states if the intersection's geometry changed
+        if (curr_light->revision_ != curr_intersection.revision)
+        {
+          RCLCPP_DEBUG_STREAM(rclcpp::get_logger("carma_wm"), "Received a new intersection geometry. intersection_id: " << (int)curr_intersection.id.id << ", and signal_group_id: " << (int)current_movement_state.signal_group);
+          traffic_signal_start_times_[curr_intersection.id.id][current_movement_state.signal_group].clear();
+          traffic_signal_states_[curr_intersection.id.id][current_movement_state.signal_group].clear();
+        }
+
+        // all maneuver types in same signal group is currently expected to share signal timing, so only 0th index is used when setting states
+        if (current_movement_state.movement_event_list.empty())
+        {
+          RCLCPP_DEBUG_STREAM(rclcpp::get_logger("carma_wm"), "Movement_event_list is empty . intersection_id: " << (int)curr_intersection.id.id << ", and signal_group_id: " << (int)current_movement_state.signal_group);
+          continue;
+        }
+        else
+        {
+          RCLCPP_DEBUG_STREAM(rclcpp::get_logger("carma_wm"), "Movement_event_list size: " << current_movement_state.movement_event_list.size() << " . intersection_id: " << (int)curr_intersection.id.id << ", and signal_group_id: " << (int)current_movement_state.signal_group);
+        }
+
+        curr_light->revision_ = curr_intersection.revision; // valid SPAT msg
+
+        auto extracted_signal_states =
+          extract_signal_states_from_movement_state(curr_intersection, current_movement_state);
+
+        traffic_signal_states_[curr_intersection.id.id][current_movement_state.signal_group] =
+          std::get<0>(extracted_signal_states);
+        traffic_signal_start_times_[curr_intersection.id.id][current_movement_state.signal_group] =
+          std::get<1>(extracted_signal_states);
+      }
+
+
+      // After all signal groups are recorded, update regulatory element objects in the map for
+      // this intersection based on whether if it is dynamic or fixed signal intersection
+      if (phase_type_ == signalized_intersection_manager::PHASE_TYPE::FIXED)
+      {
+        // TSMO UC2
+        updateSignalAsFixedSignal(curr_intersection.id.id, semantic_map);
+      }
+      else
+      {
+        // TSMO UC3
+        updateSignalAsDynamicSignal(curr_intersection.id.id, semantic_map);
+      }
+    }
+  }
+
+  lanelet::Id SignalizedIntersectionManager::getTrafficSignalId(uint16_t intersection_id, uint8_t signal_group_id)
+  {
+    lanelet::Id inter_id = lanelet::InvalId;
+    lanelet::Id signal_id = lanelet::InvalId;
+
+    if (intersection_id_to_regem_id_.find(intersection_id) != intersection_id_to_regem_id_.end())
+    {
+      inter_id = intersection_id_to_regem_id_[intersection_id];
```

Comment:
If there is no valid intersection id found, we should just throw a warning and return

Reply:
Modified

Reply:
Seems like this change might not have been pushed yet

Reply:
The change is here on my end. Please double check again the lines 538 - 542 which prints a warning and returns the invalid::Id